spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/test/java/org/springframework/cloud/function/adapter/gcloud/GcfSpringBootHttpRequestHandlerTests.java

```diff
@@ -18,270 +18,125 @@
+		when(response.getWriter()).thenReturn(new BufferedWriter(writer));
+
 		handler.service(request, response);
 
-		assertThat(response.statusCode).isEqualTo(200);
-		assertThat(writer.toString()).isEqualTo("");
-	}
-
-	@After
-	public void close() {
-		if (this.handler != null) {
-			this.handler.close();
-		}
+		assertThat(writer.toString()).isEqualTo(gson.toJson(expectedOutput));
 	}
 
 	@Configuration
-	@Import({ContextFunctionCatalogAutoConfiguration.class})
-	protected static class FunctionMessageBodyConfig {
-
+	@Import({ ContextFunctionCatalogAutoConfiguration.class })
+	protected static class HelloWorldSupplier {
 		@Bean
-		public Function<Message<Foo>, Message<Bar>> function() {
-			return (foo -> {
-				Map<String, Object> headers = new HashMap<>();
-				return new GenericMessage<>(
-					new Bar(foo.getPayload().getValue().toUpperCase()), headers);
-			});
+		public Supplier<String> supplier() {
+			return () -> "Hello World!";
 		}
-
 	}
 
 	@Configuration
-	@Import({ContextFunctionCatalogAutoConfiguration.class})
-	protected static class FunctionMessageEchoReqParametersConfig {
-
+	@Import({ ContextFunctionCatalogAutoConfiguration.class })
+	protected static class JsonInputFunction {
 		@Bean
-		public Function<Message<Void>, Message<Bar>> function() {
-			return (message -> {
-				Map<String, Object> headers = new HashMap<>();
-				headers.put("path", message.getHeaders().get("path"));
-				headers.put("query", message.getHeaders().get("query"));
-				headers.put("test-header", message.getHeaders().get("test-header"));
-				headers.put("httpMethod", message.getHeaders().get("httpMethod"));
-				return new GenericMessage<>(new Bar("body"), headers);
-			});
+		public Function<IncomingRequest, String> function() {
+			return (in) -> "Thank you for sending the message: " + in.message;
 		}
-
 	}
 
 	@Configuration
-	@Import({ContextFunctionCatalogAutoConfiguration.class})
-	protected static class FunctionMessageConsumerConfig {
-
+	@Import({ ContextFunctionCatalogAutoConfiguration.class })
+	protected static class JsonInputOutputFunction {
 		@Bean
-		public Consumer<Message<Foo>> function() {
-			return (foo -> { });
+		public Function<IncomingRequest, OutgoingResponse> function() {
+			return (in) -> new OutgoingResponse("Thank you for sending the message: " + in.message);
 		}
 	}
 
-	private static class Foo {
-
-		private String value;
-
-		Foo() {
-		}
-
-		Foo(String value) {
-			this.value = value;
-		}
-
-		public String lowercase() {
-			return this.value.toLowerCase();
-		}
-
-		public String uppercase() {
-			return this.value.toUpperCase();
-		}
-
-		public String getValue() {
-			return this.value;
-		}
-
-		public void setValue(String value) {
-			this.value = value;
+	@Configuration
+	@Import({ ContextFunctionCatalogAutoConfiguration.class })
+	protected static class JsonInputConsumer {
+		@Bean
+		public Consumer<IncomingRequest> function() {
```

Comment:
We check for `getInputType() == Void.class` in the adapter. Should we have a function that is just a supplier?

Reply:
What about `testHelloWorldSupplier`?